src/routes/animal_categories.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express');
+
+const authMiddleware = require('../middlewares/auth');
+
+const AnimalCategoriesService = require('../services/animal_categories');
+
+const AnimalPhotosService = require('../services/animal_photos');
+
+const router = express.Router();
+
+/* Get all the categories */
+router.get('/', async (req, res) => {
+  res.send(await AnimalCategoriesService.readAll());
+});
+
+/* Create new categories with photos */
+router.post('/', authMiddleware, async (req, res) => {
+  const payload = req.body;
+  Object.keys(payload).forEach(async (categoryName) => {
+    if (payload[categoryName] && payload[categoryName].length) {
+      const category = await AnimalCategoriesService.create(categoryName);
+      payload[categoryName].forEach(async (photoUrl) => {
+        await AnimalPhotosService.create({ category_id: category.id, photo_url: photoUrl });
+      });
+    }
+  });
```

Comment:
Can you see any issues with how you handle the payload here?

Reply:
Firstly, if payload is null or undefined then it can’t be accessed as arrays and it will throw error. Secondly, the way I am updating the table, I could have used bulk update. One query to update the categories table and one for photos.